deployment/src/main/java/io/quarkiverse/qute/web/deployment/QuteWebTemplateBuildItem.java

```diff
@@ -1,28 +1,31 @@
 package io.quarkiverse.qute.web.deployment;
 
-import static io.quarkiverse.qute.web.runtime.PathUtils.removeExtension;
 import static io.quarkiverse.qute.web.runtime.PathUtils.removeLeadingSlash;
 import static io.quarkiverse.qute.web.runtime.PathUtils.removeTrailingSlash;
 
+import io.quarkiverse.qute.web.runtime.PathUtils;
+import io.quarkiverse.qute.web.runtime.QuteWebBuildTimeConfig;
 import io.quarkus.builder.item.MultiBuildItem;
 
 public final class QuteWebTemplateBuildItem extends MultiBuildItem {
 
     /**
-     * templatePath is used also as path if link is null (e.g. "my-blog-post")
+     * The path relative to the template root, e.g. "my-blog-post.html". It is used also as path if link is {@code null}.
      */
     private final String templatePath;
 
     /**
-     * The link to use for this template or null to use the template path (e.g "posts/my-blog-post")
-     *
-     * If two links are identical, an exception is thrown
-     * If a link and a path are identical for different items, the link has priority
+     * The link to use for this template (e.g "posts/my-blog-post") or {@code null} to use the template path.
+     * <p>
+     * If two links are identical, an exception is thrown.
+     * <p>
+     * If a link and a path are identical for different items, the link has priority.
      */
     private final String link;
 
     public QuteWebTemplateBuildItem(String templatePath, String link) {
-        this.templatePath = removeExtension(templatePath);
+        this.templatePath = templatePath;
+        //this.templatePath = removeExtension(templatePath);
```

Comment:
it's not needed?

Reply:
Yes, not needed. I'll remove this comment...